bpf/lib/conntrack.h

```diff
@@ -434,6 +434,14 @@ static inline int __inline__ ct_lookup4(void *map, struct ipv4_ct_tuple *tuple,
 	return ret;
 }
 
+static inline void __inline__ ct_delete6(void *map, struct ipv6_ct_tuple *tuple, struct __sk_buff *skb)
+{
+	int err;
+
+	if ((err = map_delete_elem(map, tuple)) < 0)
```

Comment:
Is there a race with userspace, since both BPF and userspace may now delete entries?
Shouldn't you check here whether the returned error code indicates that the entry doesn't exist anymore, and just exit gracefully in this case?

And I'd use the same logic in userspace when deleting entries.

Reply:
Either kernel or userspace may succeed in deleting the entry. It doesn't matter. If it failed it will emit a debug trace to userspace. After that we will always exit and return a policy related drop reason as this code is only ever called when the policy is denying the packet.